pytube/contrib/search.py

```diff
@@ -0,0 +1,198 @@
+"""Module for interacting with YouTube search."""
+# Native python imports
+import logging
+
+# Local imports
+from pytube import YouTube
+from pytube.innertube import InnerTube
+
+
+logger = logging.getLogger(__name__)
+
+
+class Search:
+    def __init__(self, query):
+        self.query = query
+        self._innertube_client = InnerTube()
+
+        self._initial_results = None
+        self._results = None
```

Comment:
bit confused on different here - it is not immediately apparent

Reply:
`_initial_results` -- json object returned on first API call. continuation API calls return data that's structured differently, and that does not include some information (notably, search refinement suggestions)
`_results` -- compiled list of YouTube objects.
I'll add some comments to explain the differences.

Reply:
Ah, interesting distinction - didn't know that